everyvoice/text/features.py

```diff
@@ -75,38 +38,107 @@ def get_punctuation_features(self, tokens: list[str]) -> npt.NDArray[np.float32]
+               [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
+               [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
+               [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
+               [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1., 0.]], dtype=float32)
         """
         punctuation_features = []
         for char in tokens:
             if char == " ":
-                punctuation_features.append([1, 0, 0, 0, 0, 0, 0, 0])
+                punctuation_features.append([1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
             elif char == self.punctuation_hash["question_symbols"]:
-                punctuation_features.append([0, 1, 0, 0, 0, 0, 0, 0])
-            elif char == self.punctuation_hash["big_breaks"]:
-                punctuation_features.append([0, 0, 1, 0, 0, 0, 0, 0])
-            elif char == self.punctuation_hash["small_breaks"]:
-                punctuation_features.append([0, 0, 0, 1, 0, 0, 0, 0])
+                punctuation_features.append([0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+            elif char == self.punctuation_hash["periods"]:
+                punctuation_features.append([0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+            elif char == self.punctuation_hash["colons"]:
+                punctuation_features.append([0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0])
+            elif char == self.punctuation_hash["semi_colons"]:
+                punctuation_features.append([0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0])
+            elif char == self.punctuation_hash["commas"]:
+                punctuation_features.append([0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0])
+            elif char == self.punctuation_hash["hyphens"]:
+                punctuation_features.append([0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0])
             elif char == self.punctuation_hash["quotemarks"]:
-                punctuation_features.append([0, 0, 0, 0, 1, 0, 0, 0])
-            elif char == self.punctuation_hash["ellipsis"]:
-                punctuation_features.append([0, 0, 0, 0, 0, 1, 0, 0])
+                punctuation_features.append([0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0])
+            elif char == self.punctuation_hash["parentheses"]:
+                punctuation_features.append([0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0])
+            elif char == self.punctuation_hash["ellipses"]:
+                punctuation_features.append([0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0])
             elif char == self.punctuation_hash["exclamations"]:
-                punctuation_features.append([0, 0, 0, 0, 0, 0, 1, 0])
+                punctuation_features.append([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0])
             elif char in self.config.symbols.silence:
-                punctuation_features.append([0, 0, 0, 0, 0, 0, 0, 1])
+                punctuation_features.append([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1])
             else:
-                punctuation_features.append([0, 0, 0, 0, 0, 0, 0, 0])
+                punctuation_features.append([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
```

Comment:
I'm really not a fan of this long if/elif list. Can we not construct a dict from punc hash value to these one-hot vectors with a loop, and do a dict look up for the punc entries, followed by the last elif and else?
I mean, this code works, so you don't have to change it, but it would be nicer to refactor it.

Reply:
Hm, I kind of like just being able to see the full embedding like this - let's talk at the meeting today